opm/parser/eclipse/EclipseState/Grid/GridProperties.cpp

```diff
@@ -0,0 +1,422 @@
+            m_supportedKeywords.emplace( iter->getKeywordName(), std::move( *iter ) );
+    }
+
+
+
+    /*
+      In the case of integer properties we never really do any
+      transformation, but we have implemented this dummy int
+      specializations to enable a uniform implementation.
+    */
+
+
+    template<>
+    double GridProperties<double>::convertInputValue(const GridProperty<double>& property, double doubleValue) const {
+        const std::string& dimensionString = property.getDimensionString( );
+        return m_deckUnitSystem->getDimension(dimensionString)->getSIScaling() * doubleValue;
+    }
+
+
+    template<>
+    double GridProperties<double>::convertInputValue( double doubleValue) const {
+        return doubleValue;
+    }
+
+
+    template<>
+    int GridProperties<int>::convertInputValue(double doubleValue) const {
+        if (fabs( nearbyint( doubleValue ) - doubleValue ) < 1e-6)
+            return static_cast<int>( doubleValue );
+        else
+            throw std::invalid_argument("Expected integer argument - got: " + std::to_string( doubleValue ));
+    }
+
+    template<>
+    int GridProperties<int>::convertInputValue(const GridProperty<int>& property, double doubleValue) const {
+        return convertInputValue(doubleValue);
+    }
+
+
+    void setKeywordBox( const DeckRecord& deckRecord,
+                        BoxManager& boxManager) {
+        const auto& I1Item = deckRecord.getItem("I1");
+        const auto& I2Item = deckRecord.getItem("I2");
+        const auto& J1Item = deckRecord.getItem("J1");
+        const auto& J2Item = deckRecord.getItem("J2");
+        const auto& K1Item = deckRecord.getItem("K1");
+        const auto& K2Item = deckRecord.getItem("K2");
+
+        size_t setCount = 0;
+
+        if (!I1Item.defaultApplied(0))
+            setCount++;
+
+        if (!I2Item.defaultApplied(0))
+            setCount++;
+
+        if (!J1Item.defaultApplied(0))
+            setCount++;
+
+        if (!J2Item.defaultApplied(0))
+            setCount++;
+
+        if (!K1Item.defaultApplied(0))
+            setCount++;
+
+        if (!K2Item.defaultApplied(0))
+            setCount++;
+
+        if (setCount == 6) {
+            boxManager.setKeywordBox( I1Item.get< int >(0) - 1,
+                                      I2Item.get< int >(0) - 1,
+                                      J1Item.get< int >(0) - 1,
+                                      J2Item.get< int >(0) - 1,
+                                      K1Item.get< int >(0) - 1,
+                                      K2Item.get< int >(0) - 1);
+        } else if (setCount != 0) {
+            std::string msg = "BOX modifiers on keywords must be either "
+                "specified completely or not at all. Ignoring.";
+            throw std::invalid_argument( msg );
+            //getMessageContainer().error(deckKeyword.getFileName() + std::to_string(deckKeyword.getLineNumber()) + msg);
```

Comment:
This is past code; I considered it not worth it to get access to the filename and lineno information; but still worth keeping as a comment. I'll just remove it completely.